pointcloud/basic_octree_utils.go

```diff
@@ -23,20 +26,37 @@ func newInternalNode(tree []*BasicOctree) basicOctreeNode {
 		children: tree,
 		nodeType: internalNode,
 		point:    PointAndData{},
+		maxProb:  emptyProb,
 	}
 	return octNode
 }
 
 // Creates a new LeafNodeFilled and stores specified position and data.
 func newLeafNodeFilled(p r3.Vector, d Data) basicOctreeNode {
+	maxProb := validateData(d)
 	octNode := basicOctreeNode{
 		children: nil,
 		nodeType: leafNodeFilled,
 		point:    PointAndData{P: p, D: d},
+		maxProb:  maxProb,
 	}
 	return octNode
 }
 
+// validateData checks the data param for newLeafNodeFilled and returns a float.
+func validateData(d Data) float64 {
+	var maxProb float64
+	switch {
+	case !d.HasValue():
+		maxProb = 1
+	case d.Value() > 100 || d.Value() < 0:
```

Comment:
[nit] might be a bit clearer if this is `d.Value() <= 100 && d.Value() >= 0` so the default can setting the `emptyProb`

Reply:
cannot do this.
sps 101
then 101 <= 100 is false but 101 >= 0